### ISSUE TITLE
Extend LXD marketplace import procedure with supported distros

### ISSUE THREAD

**Issue by dann1** [OpenNebula/one#3178]
**Description**
Available images on [the public LXD image server that backs the LXD marketplace](https://images.linuxcontainers.org) matching the [context supported OS](https://github.com/OpenNebula/addon-context-linux#tested-platforms) should all be handled by the import procedure.

**Use case**
Auto-contextualize apps when importing them.

[context supported apps](https://github.com/OpenNebula/addon-context-linux#tested-platforms)

## Progress Status

AlmaLinux
- [x] 8

Alpine Linux
- [x] 3.11
- [x] 3.12
- [x] 3.13
- [x] 3.14

ALT Linux
- [x] p9
- [x] p10
- [x] Sisyphus

Amazon Linux
- [x] 2

CentOS
- [x] 7
- [x] 8
- [x] 8 Stream

Debian
- [x] 8
- [x] 9
- [x] 10
- [x] 11

Devuan
- [x] 2

Fedora
- [x] 32
- [x] 33
- [x] 34

openSUSE
- [x] 15
- [ ] Tumbleweed

Oracle Linux
- [x] 7
- [x] 8

Rocky Linux
- [x] 8

Springdale Linux
- [x] 8

Ubuntu
- [x] 14.04
- [x] 16.04
- [x] 18.04
- [x] 20.04
- [x] 21.04
- [x] 21.10
- [x] 22.04

**Comment by dann1** [comment]
Fedora 30 fails with hostname contextualization

```
[root@LXCNAME ~]# hostnamectl set-hostname --static new_hostname
Failed to create bus connection: Connection refused
[root@LXCNAME ~]# systemctl list-units
  UNIT                                   LOAD   ACTIVE SUB       DESCRIPTION                                          
  -.mount                                loaded active mounted   /                                                    
  context.mount                          loaded active mounted   /context                                             
  dev-.lxd\x2dmounts.mount               loaded active mounted   /dev/.lxd-mounts                                     
  dev-fuse.mount                         loaded active mounted   /dev/fuse                                            
  dev-lxd.mount                          loaded active mounted   /dev/lxd                                             
  dev-mqueue.mount                       loaded active mounted   POSIX Message Queue File System                      
  dev-net-tun.mount                      loaded active mounted   /dev/net/tun                                         
  dev-ptmx.mount                         loaded active mounted   /dev/ptmx                                            
  proc-cpuinfo.mount                     loaded active mounted   /proc/cpuinfo                                        
  proc-diskstats.mount                   loaded active mounted   /proc/diskstats                                      
  proc-meminfo.mount                     loaded active mounted   /proc/meminfo                                        
  proc-stat.mount                        loaded active mounted   /proc/stat                                           
  proc-swaps.mount                       loaded active mounted   /proc/swaps                                          
  proc-sys-fs-binfmt_misc.mount          loaded active mounted   Arbitrary Executable File Formats File System        
  proc-sys-net.mount                     loaded active mounted   /proc/sys/net                                        
  proc-sysrq\x2dtrigger.mount            loaded active mounted   /proc/sysrq-trigger                                  
  proc-uptime.mount                      loaded active mounted   /proc/uptime                                         
  sys-devices-virtual-net.mount          loaded active mounted   /sys/devices/virtual/net                             
● sys-fs-fuse-connections.mount          masked active mounted   /sys/fs/fuse/connections                             
  sys-kernel-debug.mount                 loaded active mounted   Kernel Debug File System                             
  tmp.mount                              loaded active mounted   Temporary Directory (/tmp)                           
  systemd-ask-password-console.path      loaded active waiting   Dispatch Password Requests to Console Directory Watch
  systemd-ask-password-wall.path         loaded active waiting   Forward Password Requests to Wall Directory Watch    
  init.scope                             loaded active running   System and Service Manager                           
● dbus-broker.service                    loaded failed failed    D-Bus System Message Bus                             
  ldconfig.service                       loaded active exited    Rebuild Dynamic Linker Cache                         
  one-context-local.service              loaded active exited    OpenNebula pre-networking contextualization script   
  one-context.service                    loaded active exited    OpenNebula contextualization script                  
  selinux-autorelabel-mark.service       loaded active exited    Mark the need to relabel after reboot                
  sshd.service                           loaded active running   OpenSSH server daemon                                
  systemd-hwdb-update.service            loaded active exited    Rebuild Hardware Database                            
  systemd-journal-catalog-update.service loaded active exited    Rebuild Journal Catalog                              
  systemd-journal-flush.service          loaded active exited    Flush Journal to Persistent Storage                  
  systemd-journald.service               loaded active running   Journal Service                                      
● systemd-networkd.service               loaded failed failed    Network Service                                      
● systemd-resolved.service               loaded failed failed    Network Name Resolution                              
  systemd-sysctl.service                 loaded active exited    Apply Kernel Variables                               
  systemd-sysusers.service               loaded active exited    Create System Users                                  
  systemd-tmpfiles-setup-dev.service     loaded active exited    Create Static Device Nodes in /dev                   
  systemd-tmpfiles-setup.service         loaded active exited    Create Volatile Files and Directories                
  systemd-update-done.service            loaded active exited    Update is Completed                                  
  systemd-update-utmp.service            loaded active exited    Update UTMP about System Boot/Shutdown               
  systemd-user-sessions.service          loaded active exited    Permit User Sessions                                 
  -.slice                                loaded active active    Root Slice                                           
  system-sshd\x2dkeygen.slice            loaded active active    system-sshd\x2dkeygen.slice                          
[root@LXCNAME ~]# systemctl status dbus-broker
● dbus-broker.service - D-Bus System Message Bus
   Loaded: loaded (/usr/lib/systemd/system/dbus-broker.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2019-07-09 06:19:38 UTC; 48s ago
     Docs: man:dbus-broker-launch(1)
  Process: 611 ExecStart=/usr/bin/dbus-broker-launch --scope system --audit (code=exited, status=226/NAMESPACE)
 Main PID: 611 (code=exited, status=226/NAMESPACE)

Jul 09 06:19:38 LXCNAME systemd[1]: Failed to start D-Bus System Message Bus.
Jul 09 06:19:38 LXCNAME systemd[1]: Starting D-Bus System Message Bus...
Jul 09 06:19:38 LXCNAME systemd[611]: dbus-broker.service: Failed to set up mount namespacing: Permission denied
Jul 09 06:19:38 LXCNAME systemd[611]: dbus-broker.service: Failed at step NAMESPACE spawning /usr/bin/dbus-broker-launch: Permission denied
Jul 09 06:19:38 LXCNAME systemd[1]: dbus-broker.service: Main process exited, code=exited, status=226/NAMESPACE
Jul 09 06:19:38 LXCNAME systemd[1]: dbus-broker.service: Failed with result 'exit-code'.
Jul 09 06:19:38 LXCNAME systemd[1]: Failed to start D-Bus System Message Bus.
Jul 09 06:19:38 LXCNAME systemd[1]: dbus-broker.service: Start request repeated too quickly.
Jul 09 06:19:38 LXCNAME systemd[1]: dbus-broker.service: Failed with result 'exit-code'.
Jul 09 06:19:38 LXCNAME systemd[1]: Failed to start D-Bus System Message Bus.
[root@LXCNAME ~]# systemctl restart dbus
Job for dbus-broker.service failed because the control process exited with error code.
See "systemctl status dbus-broker.service" and "journalctl -xe" for details.
[root@LXCNAME ~]# journalctl -xe
-- Logs begin at Tue 2019-07-09 06:00:25 UTC, end at Tue 2019-07-09 06:20:37 UTC. --
Jul 09 06:00:25 LXCNAME systemd-journald[41]: Journal started
-- Subject: The journal has been started
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- The system journal process has started up, opened the journal
-- files for writing and is now ready to process requests.
Jul 09 06:00:25 LXCNAME systemd-journald[41]: Runtime journal (/run/log/journal/e555d950bb404b898d83651e9c4def52) is 8.0M, max 151.1M, 143.1M free.
-- Subject: Disk space used by the journal
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- Runtime journal (/run/log/journal/e555d950bb404b898d83651e9c4def52) is currently using 8.0M.
-- Maximum allowed usage is set to 151.1M.
-- Leaving at least 226.7M free (of currently available 1.4G of disk space).
-- Enforced usage limit is thus 151.1M, of which 143.1M are still available.
-- 
-- The limits controlling how much disk space is used by the journal may
-- be configured with SystemMaxUse=, SystemKeepFree=, SystemMaxFileSize=,
-- RuntimeMaxUse=, RuntimeKeepFree=, RuntimeMaxFileSize= settings in
-- /etc/systemd/journald.conf. See journald.conf(5) for details.
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '0' to 'kernel/yama/ptrace_scope', ignoring: Read-only file system
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '|/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h %e' to 'kernel/core_pattern', ignoring: Read-only fi
le system
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '16' to 'kernel/sysrq', ignoring: Read-only file system
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '1' to 'kernel/core_uses_pid', ignoring: Read-only file system
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write 'fq_codel' to 'net/core/default_qdisc', ignoring: No such file or directory
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '1' to 'fs/protected_hardlinks', ignoring: Read-only file system
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '1' to 'fs/protected_symlinks', ignoring: Read-only file system
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '1' to 'fs/protected_regular', ignoring: No such file or directory
Jul 09 06:00:25 LXCNAME systemd-sysctl[39]: Couldn't write '1' to 'fs/protected_fifos', ignoring: No such file or directory
Jul 09 06:00:25 LXCNAME systemd[1]: Starting Flush Journal to Persistent Storage...
-- Subject: A start job for unit systemd-journal-flush.service has begun execution
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
-- 
-- A start job for unit systemd-journal-flush.service has begun execution.
-- 
-- The job identifier is 17.
Jul 09 06:00:25 LXCNAME systemd-journald[41]: Time spent on flushing to /var is 1.332ms for 12 entries.
Jul 09 06:00:25 LXCNAME systemd-journald[41]: System journal (/var/log/journal/e555d950bb404b898d83651e9c4def52) is 8.0M, max 97.5M, 89.5M free.
-- Subject: Disk space used by the journal
-- Defined-By: systemd
-- Support: https://lists.freedesktop.org/mailman/listinfo/systemd-devel
```

**Comment by dann1** [comment]
[Installing curl and openssh on opensuse images](https://github.com/OpenNebula/one/commit/21b5c42cfbe4c3f402d428c8a7d091981635d2af) fails during chroot

opensuse15

```
Repository 'openSUSE-Leap-15.0-Non-Oss' is invalid.
[repo-non-oss|http://download.opensuse.org/distribution/leap/15.0/repo/non-oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'openSUSE-Leap-15.0-Oss' is invalid.
[repo-oss|http://download.opensuse.org/distribution/leap/15.0/repo/oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'openSUSE-Leap-15.0-Update' is invalid.
[repo-update|http://download.opensuse.org/update/leap/15.0/oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'openSUSE-Leap-15.0-Update-Non-Oss' is invalid.
[repo-update-non-oss|http://download.opensuse.org/update/leap/15.0/non-oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Some of the repositories have not been refreshed because of an error.
No provider of 'curl' found.
No provider of 'openssh' found.
Retrieving repository 'openSUSE-Leap-15.0-Non-Oss' metadata [..error]
Warning: Skipping repository 'openSUSE-Leap-15.0-Non-Oss' because of the above error.
Retrieving repository 'openSUSE-Leap-15.0-Oss' metadata [.error]
Warning: Skipping repository 'openSUSE-Leap-15.0-Oss' because of the above error.
Retrieving repository 'openSUSE-Leap-15.0-Update' metadata [.error]
Warning: Skipping repository 'openSUSE-Leap-15.0-Update' because of the above error.
Retrieving repository 'openSUSE-Leap-15.0-Update-Non-Oss' metadata [..error]
Warning: Skipping repository 'openSUSE-Leap-15.0-Update-Non-Oss' because of the above error.
Loading repository data...
Reading installed packages...
'curl' not found in package names. Trying capabilities.
'openssh' not found in package names. Trying capabilities.
```

opensuse42

```
Repository 'NON OSS' is invalid.
[NON OSS|http://download.opensuse.org/distribution/leap/42.3/repo/non-oss/suse/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'NON OSS Update' is invalid.
[NON OSS Update|http://download.opensuse.org/update/leap/42.3/non-oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'OSS' is invalid.
[OSS|http://download.opensuse.org/distribution/leap/42.3/repo/oss/suse/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'OSS Update' is invalid.
[OSS Update|http://download.opensuse.org/update/leap/42.3/oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Some of the repositories have not been refreshed because of an error.
No provider of 'curl' found.
No provider of 'openssh' found.
Retrieving repository 'NON OSS' metadata [.
Warning: File 'repomd.xml' from repository 'NON OSS' is signed with an unknown key ''.

    Note: Signing data enables the recipient to verify that no modifications occurred after the data
    were signed. Accepting data with no, wrong or unknown signature can lead to a corrupted system
    and in extreme cases even to a system compromise.

    Note: File 'repomd.xml' is the repositories master index file. It ensures the integrity of the
    whole repo.

    Warning: We can't verify that no one meddled with this file, so it might not be trustworthy
    anymore! You should not continue unless you know it's safe.

File 'repomd.xml' from repository 'NON OSS' is signed with an unknown key ''. Continue? [yes/no] (no): no
error]
Warning: Skipping repository 'NON OSS' because of the above error.
Retrieving repository 'NON OSS Update' metadata [..
Warning: File 'repomd.xml' from repository 'NON OSS Update' is signed with an unknown key ''.

    Note: Signing data enables the recipient to verify that no modifications occurred after the data
    were signed. Accepting data with no, wrong or unknown signature can lead to a corrupted system
    and in extreme cases even to a system compromise.

    Note: File 'repomd.xml' is the repositories master index file. It ensures the integrity of the
    whole repo.

    Warning: We can't verify that no one meddled with this file, so it might not be trustworthy
    anymore! You should not continue unless you know it's safe.

File 'repomd.xml' from repository 'NON OSS Update' is signed with an unknown key ''. Continue? [yes/no] (no): no
error]
Warning: Skipping repository 'NON OSS Update' because of the above error.
Retrieving repository 'OSS' metadata [.
Warning: File 'repomd.xml' from repository 'OSS' is signed with an unknown key ''.

    Note: Signing data enables the recipient to verify that no modifications occurred after the data
    were signed. Accepting data with no, wrong or unknown signature can lead to a corrupted system
    and in extreme cases even to a system compromise.

    Note: File 'repomd.xml' is the repositories master index file. It ensures the integrity of the
    whole repo.

    Warning: We can't verify that no one meddled with this file, so it might not be trustworthy
    anymore! You should not continue unless you know it's safe.

File 'repomd.xml' from repository 'OSS' is signed with an unknown key ''. Continue? [yes/no] (no): no
error]
Warning: Skipping repository 'OSS' because of the above error.
Retrieving repository 'OSS Update' metadata [.
Warning: File 'repomd.xml' from repository 'OSS Update' is signed with an unknown key ''.

    Note: Signing data enables the recipient to verify that no modifications occurred after the data
    were signed. Accepting data with no, wrong or unknown signature can lead to a corrupted system
    and in extreme cases even to a system compromise.

    Note: File 'repomd.xml' is the repositories master index file. It ensures the integrity of the
    whole repo.

    Warning: We can't verify that no one meddled with this file, so it might not be trustworthy
    anymore! You should not continue unless you know it's safe.

File 'repomd.xml' from repository 'OSS Update' is signed with an unknown key ''. Continue? [yes/no] (no): no
error]
Warning: Skipping repository 'OSS Update' because of the above error.
Loading repository data...
Reading installed packages...
'curl' not found in package names. Trying capabilities.
'openssh' not found in package names. Trying capabilities.
```

tumbleweed

```
Repository 'openSUSE-Tumbleweed-Non-Oss' is invalid.
[repo-non-oss|http://download.opensuse.org/tumbleweed/repo/non-oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'openSUSE-Tumbleweed-Oss' is invalid.
[repo-oss|http://download.opensuse.org/tumbleweed/repo/oss/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Repository 'openSUSE-Tumbleweed-Update' is invalid.
[repo-update|http://download.opensuse.org/update/tumbleweed/] Valid metadata not found at specified URL
Please check if the URIs defined for this repository are pointing to a valid repository.
Some of the repositories have not been refreshed because of an error.
No provider of 'curl' found.
No provider of 'openssh' found.
Retrieving repository 'openSUSE-Tumbleweed-Non-Oss' metadata [.error]
Warning: Skipping repository 'openSUSE-Tumbleweed-Non-Oss' because of the above error.
Retrieving repository 'openSUSE-Tumbleweed-Oss' metadata [.error]
Warning: Skipping repository 'openSUSE-Tumbleweed-Oss' because of the above error.
Retrieving repository 'openSUSE-Tumbleweed-Update' metadata [.error]
Warning: Skipping repository 'openSUSE-Tumbleweed-Update' because of the above error.
Loading repository data...
Reading installed packages...
'curl' not found in package names. Trying capabilities.
'openssh' not found in package names. Trying capabilities.
```
